oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -578,10 +588,70 @@ private boolean visitTerm(String propertyName, String text, String boost, boolea
                 return true;
             }
         });
-        
+
         return Query.of(q -> q.bool(result.get()));
     }
 
+    private ObjectBuilder<BoolQuery> inference(BoolQuery.Builder b, String propertyName, String text, PlanResult pr, boolean dbEnabled) {
+        ElasticIndexDefinition.InferenceDefinition.Query q = null;
+        // select first query eligible for the given text
+        // TODO: evaluate if/how to handle multiple queries
+        String  queryText = text;
+        for (ElasticIndexDefinition.InferenceDefinition.Query query : elasticIndexDefinition.inferenceDefinition.queries) {
+            if (query.isEligibleForInput(queryText)) {
+                queryText = query.rewrite(queryText);
+                if (query.hasMinTerms(queryText)) {
+                    q = query;
+                    break;
+                }
+            }
+        }
+
+        QueryStringQuery.Builder qsqBuilder = fullTextQuery(queryText, getElasticFieldName(propertyName), pr, dbEnabled);
+
+        if (q != null) {
```

Comment:
Could you write a comment that just briefly explains when the inference query could be null and what happens when that is the case? Looks like it will just do a basic fulltext search, but am not 100% sure.